service/matching/task_queue_partition_manager.go

```diff
@@ -483,25 +482,22 @@ func (pm *taskQueuePartitionManagerImpl) Describe(
 			buildIds[k] = true
 		}
 	}
+	pm.versionedQueuesLock.RUnlock()
```

Comment:
`defer` has been removed here since the call to pm.getPhysicalQueue results in using a Write Lock (in the event that a versioned queue was not created at all). Thus, the read lock was released after reading `versionedQueues` and updating `buildIds`.